stdlib/solana.c

```diff
@@ -50,14 +43,12 @@ entrypoint(const uint8_t *input)
         }
     }
 
-    if (params.ka_cur != 0)
-    {
-        return ERROR_INVALID_INSTRUCTION_DATA;
-    }
-
     __init_heap();
 
-    return solang_dispatch(&params);
+    // solang_dispatch() is generated in codegen. In codegen, we do not have access
+    // the params. We could use Builtin::Calldata for the input but this copies,
+    // so we simply pass in the input
```

Comment:
```suggestion
    // solang_dispatch() is generated in codegen, where we do not have access
    // the params. We could use Builtin::Calldata for the input but this copies,
    // so we simply pass in the input
```

Reply:
Isn't it possible to use `Expression::MemberAccess`?

Reply:
You can do `Expression::StructMember(Loc::Codegen, Type::Struct(StructType::SolParamenters), Type::Ref(Type::BufferPointer.into), 3)`

Reply:
Very nice, I had missed that we have `StructType::SolParameters` . Fixed now.